foundation-model-stack/multi-nic-cni
# Add status to MultiNicNetwork CR and fix some bugs

The main object of this PR is to add CIDR and route status regarding MultiNicNetwork.
```go
// +enum
type RouteStatus string

const (
	// AllRouteApplied indicates that all routes are applied
	AllRouteApplied RouteStatus = "Success"

	// ApplyingRoute indicates that the new cidr is just recomputed and waiting for route update
	ApplyingRoute RouteStatus = "WaitForRoutes"

	// SomeRouteFailed indicates that some route cannot be applied, need attention
	SomeRouteFailed RouteStatus = "Failed"

	// RouteUnknown indicates that some daemon cannot be connected
	RouteUnknown RouteStatus = "Unknown"

	// RouteNoApplied indicates that there is no L3 configuration applied
	RouteNoApplied RouteStatus = "N/A"
)

type NicNetworkResult struct {
	NetAddress string `json:"netAddress"`
	NumOfHost  int    `json:"numOfHosts"`
}

// MultiNicNetworkStatus defines the observed state of MultiNicNetwork
type MultiNicNetworkStatus struct {
	ComputeResults []NicNetworkResult `json:"computeResults"`
	Status         RouteStatus        `json:"routeStatus"`
	LastSyncTime   metav1.Time        `json:"lastSyncTime"`
}
```

Additionally, I improved the code with the following fixes.
- check daemon pod container ready state instead of ip assignment to reduce probability of connection refuse failure when try connecting to daemon at the first time
- remove deprecated CIDR namespace (now cluster-wide)
- update test suite to avoid conflicts among test cases with multinicnetwork deployment
- shrink too long name in connection-check program
- reduce hostinterface reconcile time when getting connection refuse failure
